src/registrar/admin.py

```diff
@@ -2567,19 +2745,96 @@ def queryset(self, request, queryset):
+            
+    class GenericOrgFilter(admin.SimpleListFilter):
+        """Custom Generic Organization filter that accomodates portfolio feature.
+        If we have a portfolio, use the portfolio's organization.  If not, use the
+        organization in the Domain Information object."""
+
+        title = "generic organization"
+        parameter_name = 'converted_generic_orgs'
 
+        def lookups(self, request, model_admin):
+            converted_generic_orgs = set()
+
+            for domainInfo in DomainInformation.objects.all():
+                converted_generic_org = domainInfo.converted_generic_org_type
+                if converted_generic_org:
+                    converted_generic_orgs.add(converted_generic_org)
+            
+            return sorted((org, org) for org in converted_generic_orgs)
+
+        # Filter queryset
+        def queryset(self, request, queryset):
+            if self.value():  # Check if a generic org is selected in the filter
+                return queryset.filter(
+                    # Filter based on the generic org value returned by converted_generic_org_type
+                    id__in=[
+                        domain.id for domain in queryset if domain.domain_info.converted_generic_org_type and domain.domain_info.converted_generic_org_type == self.value()
+                    ]
+                )
+            return queryset
+        
+    class FederalTypeFilter(admin.SimpleListFilter):
+        """Custom Federal Type filter that accomodates portfolio feature.
+        If we have a portfolio, use the portfolio's federal type.  If not, use the
+        federal type in the Domain Information object."""
+
+        title = "federal type"
+        parameter_name = 'converted_federal_types'
+
+        def lookups(self, request, model_admin):
+            converted_federal_types = set()
+            # converted_federal_types.add("blah")
+
+            for domainInfo in DomainInformation.objects.all():
+                converted_federal_type = domainInfo.converted_federal_type
+                if converted_federal_type:
+                    converted_federal_types.add(converted_federal_type)
+            
+            return sorted((fed, fed) for fed in converted_federal_types)
+
+        # Filter queryset
+        def queryset(self, request, queryset):
+            if self.value():  # Check if a generic org is selected in the filter
+                return queryset.filter(
+                    # Filter based on the generic org value returned by converted_generic_org_type
+                    id__in=[
+                        domain.id for domain in queryset if domain.domain_info.converted_federal_type and domain.domain_info.converted_federal_type == self.value()
+                    ]
+                )
+            return queryset
+
+        
+    def get_queryset(self, request):
```

Comment:
Why was this moved up? The typical convention I have seen is Class definition, followed by all class variables, then sub classes followed by class functions. We already have this format being broken by defining some variables above sub classes but I would prefer we not make the format even more outside of convention by moving functions up